structmatcher/structmatcher.go

```diff
@@ -87,3 +87,48 @@ func ExpectStructsToMatchIncluding(expected interface{}, actual interface{}, inc
 		Fail(strings.Join(mismatches, "\n"))
 	}
 }
+
+type Matcher struct {
+	expected        interface{}
+	includingFields []string
+	excludingFields []string
+	mismatches      []string
+}
+
+var _ types.GomegaMatcher = &Matcher{}
+
+func Match(expected interface{}) *Matcher {
```

Comment:
It's a bit confusing to have both this and the other `Match` function called the same thing. Can we think of another name? MatchStruct?

Or maybe we could make separate matchers for `MatchIncluding` or `MatchExcluding`?

I'm looking at [gstruct](https://github.com/onsi/gomega/tree/master/gstruct) for examples

Reply:
Yeah, the namespace collision there is a bit annoying.  `MatchStruct` is probably the simplest change, but `structmatcher.MatchStruct` has a bit of a stutter.  Splitting it into `MatchIncluding`, `MatchExcluding`, and `MatchExactly` would also work.  No strong opinions on which approach to take.